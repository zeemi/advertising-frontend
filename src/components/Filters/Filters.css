.Filters {
    width: 100%;
    height: 100%;
}

.Filters .button-wrapper {
    margin-top: 50px;
    text-align: center;
}

.Filters button {
    background-color: #4CAF50; /* Green */
    border: none;
    color: white;
    padding: 15px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
}