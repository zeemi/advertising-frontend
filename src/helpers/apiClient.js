import settings from "../settings";

export default class ApiClient {
  static get(path){
    return fetch(`${settings.backendUrl}${path}`).then(res=> res.json());
  }

  static getMetrics() {
    return this.get('/metrics')
  }
}
