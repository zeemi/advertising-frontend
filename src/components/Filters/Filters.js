import React, {useEffect} from 'react';
import apiClient from "../../helpers/apiClient";


const Filters = ({isFetching, setIsFetching, setMetrics}) => {
  const fetchMetrics = async () => {
    setIsFetching(true);
    const metrics = await apiClient.getMetrics([], []);
    setMetrics(metrics);
    setIsFetching(false);
  };

  useEffect(() => {
    fetchMetrics()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return <div>
    Filters
    <button onClick={fetchMetrics} disabled={isFetching}>Apply</button>
  </div>
};

export default Filters;