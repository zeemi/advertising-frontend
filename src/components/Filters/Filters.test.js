import React from 'react';
import {mount, shallow} from 'enzyme';
import Filters from './Filters';
import apiClient from '../../helpers/apiClient';


jest.mock('../../helpers/apiClient', () => ({getMetrics: jest.fn()}));

describe('<Filters>', function () {
  beforeEach(function () {
    apiClient.getMetrics.mockReset();
  });

  it('should render apply button', function () {
    expect(shallow(<Filters/>)).toMatchSnapshot()
  });

  it('should call fetch metrics', function () {
    const wrapper = mount(<Filters setIsFetching={()=>{}} setMetrics={()=>{}}/>);
    expect(apiClient.getMetrics).toHaveBeenCalledWith([], [])
  });

  it('should call fetch metrics on button "apply" click', function () {
    const wrapper = mount(<Filters setIsFetching={()=>{}} setMetrics={()=>{}}/>);
    apiClient.getMetrics.mockReset();
    wrapper.find('button').simulate('click');
    expect(apiClient.getMetrics).toHaveBeenCalledWith([], [])
  });

});