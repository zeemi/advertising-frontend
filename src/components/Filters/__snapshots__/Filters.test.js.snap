// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters> should render apply button 1`] = `
<div>
  Filters
  <button
    onClick={[Function]}
  >
    Apply
  </button>
</div>
`;
